cli/pkg/validate/required_keys.go

```diff
@@ -92,5 +112,247 @@ func (rk *RequiredKeysValidator) Validate(dc *catalog.DataCatalog) []ValidationE
 		}
 	}
 
+	// Custom Types required keys
+	for group, customTypes := range dc.CustomTypes {
+		for _, customType := range customTypes {
+			reference := fmt.Sprintf("#/custom-types/%s/%s", group, customType.LocalID)
+
+			// Check mandatory fields
+			if customType.LocalID == "" || customType.Name == "" || customType.Type == "" {
+				errors = append(errors, ValidationError{
+					error:     fmt.Errorf("id, name and type fields on custom type are mandatory"),
+					Reference: reference,
+				})
+				continue
+			}
+
+			// Check if config is present
+			if customType.Config == nil {
+				errors = append(errors, ValidationError{
+					error:     fmt.Errorf("config field is mandatory on custom type"),
+					Reference: reference,
+				})
+			}
+
+			// If type is object, properties is required
+			if customType.Type == "object" && len(customType.Properties) == 0 {
+				errors = append(errors, ValidationError{
+					error:     fmt.Errorf("properties array is required for custom type with type 'object'"),
+					Reference: reference,
+				})
+			}
+
+			// Check each property in properties has id field
+			if customType.Type == "object" {
+				for i, prop := range customType.Properties {
+					if prop.ID == "" {
+						errors = append(errors, ValidationError{
+							error:     fmt.Errorf("id field is mandatory for property at index %d in custom type", i),
+							Reference: reference,
+						})
+					}
+				}
+			}
+
+			// Name format validation - no need to check if name is empty as we already checked above
+			if !customTypeNameRegex.MatchString(customType.Name) {
+				errors = append(errors, ValidationError{
+					error:     fmt.Errorf("custom type name must start with a capital letter and contain only letters, numbers, underscores and dashes, and be between 2 and 65 characters long"),
+					Reference: reference,
+				})
+			}
+
+			// Type validation
+			if !slices.Contains(validTypes, customType.Type) {
+				errors = append(errors, ValidationError{
+					error:     fmt.Errorf("invalid data type. Acceptable values are string, number, integer, boolean, null, array, and object"),
+					Reference: reference,
+				})
+			}
+
+			// Type-specific config validations
+			if customType.Config != nil {
```

Comment:
Should we be able to apply the same validations in properties' propConfig?

Reply:
It's a good point, I'll verify and fix it.

Reply:
@fxenik although I am tempted to use the same config check for properties as well, does it make sense to do that in a separate PR to separate the concerns ?

If yes, I can create another PR to handle this change in a more DRY friendly way.

Reply:
Yes we don't have to do it now if you think it needs more thought. I was just thinking about what we discussed in the past, that the CLI should be able to validate propConfig even before making any API calls. We can have this as a backlog item and see if we can reuse the type checking that's introduced here.